pkg/services/ngalert/schedule/registry.go

```diff
@@ -315,6 +315,7 @@ func (r ruleWithFolder) Fingerprint() fingerprint {
 	// fields that do not affect the state.
 	// TODO consider removing fields below from the fingerprint
 	writeInt(int64(rule.For))
+	writeInt(int64(rule.KeepFiringFor))
```

Comment:
I am leaning towards not including it into the fingerprint as it does not affect the state consistency.

Reply:
Makes sense, I also added it under "TODO consider removing fields below from the fingerprint", but let's not add it at all 😄